File: backend/src/main/java/com/examplatform/result/Result.java
package com.examplatform.result;

import com.examplatform.exam.Exam;
import com.examplatform.submission.Submission;
import com.examplatform.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "results")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Result {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "submission_id", nullable = false, unique = true)
    private Submission submission;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private User student;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exam_id", nullable = false)
    private Exam exam;

    @Column(name = "total_questions", nullable = false)
    private Integer totalQuestions;

    @Column(name = "correct_answers", nullable = false)
    private Integer correctAnswers;

    @Column(nullable = false)
    private Double score;

    @Column(name = "is_passed", nullable = false)
    private Boolean isPassed;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/types/Result.ts
export interface Result {
  id: number;
  submissionId: number;
  studentId: number;
  studentName: string;
  studentEmail: string;
  examId: number;
  examTitle: string;
  totalQuestions: number;
  correctAnswers: number;
  score: number;
  passScore: number;
  isPassed: boolean;
  createdAt: string;
}

export interface SubmitAnswerDto {
  questionId: number;
  selectedAnswerId?: number;
}

export interface SubmitExamDto {
  examId: number;
  answers: SubmitAnswerDto[];
}
